File: official-docs/examples/examples/python-server-wallets/server_wallet/delegated/sign_message.py
"""
Delegated Wallet Message Signing Demo

Sign messages using a delegated wallet for authentication and verification.

## Prerequisites

This script requires a wallet.json file with delegated access credentials.
See wallet.json.example for the required format.

The delegated share is obtained through a separate process where the user
grants your application permission to sign on their behalf.

## Usage

  python -m server_wallet.delegated.sign_message "Hello, World!"

## Use Cases

- Authenticate users by proving wallet ownership
- Sign authorization tokens or session data
- Verify identity without on-chain transactions
- Create off-chain signatures for gasless flows
"""

import argparse
import json
from datetime import datetime
from pathlib import Path

from lib.cli import run_script
from lib.dynamic import delegated_evm_client, delegated_sign_message

WALLET_FILE = Path(__file__).parent / "wallet.json"


async def sign_message(message: str):
    # Load delegated wallet credentials
    if not WALLET_FILE.exists():
        raise SystemExit(
            "wallet.json not found in server_wallet/delegated/ directory.\n"
            "See wallet.json.example for the required format.\n"
            "See README.md for how to obtain delegated access credentials."
        )

    wallet = json.loads(WALLET_FILE.read_text())

    # Step 1: Create delegated client
    client = await delegated_evm_client()

    print("\nSigning message...")
    start = datetime.now()

    # Step 2: Sign using the wallet owner's delegated share
    signature = await delegated_sign_message(
        client,
        wallet_id=wallet["walletId"],
        wallet_api_key=wallet["walletApiKey"],
        key_share=wallet["delegatedShare"],
        message=message,
        chain_name="EVM",
        is_formatted=False,
    )

    # Step 3: Display results
    duration = (datetime.now() - start).total_seconds()
    print(f"\nMessage signed in {duration:.2f}s")
    print(f"Message: \"{message}\"")
    print(f"Signature: {signature}")

    return signature


def main():
    parser = argparse.ArgumentParser(description="Sign messages with delegated wallets")
    parser.add_argument("message", help="Message to sign")

    args = parser.parse_args()

    run_script(lambda: sign_message(args.message))


if __name__ == "__main__":
    main()

File: official-docs/examples/examples/nodejs-server-wallets/src/delegated/sign-message.ts
#!/usr/bin/env tsx

/**
 * Delegated Wallet Message Signing Demo
 *
 * Sign messages using a delegated wallet for authentication and verification.
 *
 * ## Prerequisites
 *
 * This script requires a wallet.json file with delegated access credentials.
 * See wallet.json.example for the required format.
 *
 * The delegated share is obtained through a separate process where the user
 * grants your application permission to sign on their behalf.
 *
 * ## Usage
 *
 *   pnpm delegated:sign-msg "Hello, World!"
 *
 * ## Use Cases
 *
 * - Authenticate users by proving wallet ownership
 * - Sign authorization tokens or session data
 * - Verify identity without on-chain transactions
 * - Create off-chain signatures for gasless flows
 */

import { parseArgs, runScript } from "../lib/cli";
import { delegatedEvmClient, delegatedSignMessage } from "../lib/dynamic";
import wallet from "./wallet.json";

/**
 * Step 1: Sign a message with delegated credentials
 *
 * Unlike server wallets where you control the key shares,
 * delegated wallets use credentials provided by the wallet owner.
 */
async function signMessage(message: string) {
  // Create delegated client
  const client = delegatedEvmClient();

  console.info(`\nSigning message...`);
  const start = Date.now();

  // Sign using the wallet owner's delegated share
  const signature = await delegatedSignMessage(client, {
    walletId: wallet.walletId,
    walletApiKey: wallet.walletApiKey,
    keyShare: wallet.delegatedShare,
    message,
  });

  // Step 2: Display results
  const duration = ((Date.now() - start) / 1000).toFixed(2);
  console.info(`\nMessage signed in ${duration}s`);
  console.info(`Message: "${message}"`);
  console.info(`Signature: ${signature}`);

  return signature;
}

runScript(async () => {
  const { positional } = parseArgs(process.argv);
  const message = positional[0];

  if (!message) {
    console.error("Please provide a message to sign");
    console.error("\nUsage:");
    console.error('  pnpm delegated:sign-msg "Hello, World!"');
    process.exit(1);
  }

  await signMessage(message);
});
